Components/Pages/Login.js

```diff
@@ -1,49 +1,51 @@
-
 import React, { useState } from "react";
 import { AsyncStorage, Dimensions, StyleSheet, Text } from "react-native";
 import { LOGIN_URI } from "react-native-dotenv";
-import { Snackbar } from 'react-native-paper';
+import { Snackbar } from "react-native-paper";
 import { TextField } from "../Atoms/Fields";
 import SignUpContainer from "../Molecules/SignUpContainer";
 import axios from "axios";
-import moment from "moment"
+import moment from "moment";
 
 const Login = ({ navigation }) => {
+  const [loading, updateLoading] = useState(false);
+
   const [password, updatePassword] = useState("");
   const [email, updateEmail] = useState("");
   const [visible, setVisible] = useState(0);
   const [message, setMessage] = useState("");
 
-  const _onDismissSnackBar = () => { setVisible(0), setMessage("") };
+  const _onDismissSnackBar = () => {
+    setVisible(0), setMessage("");
+  };
 
   const onLoginSubmit = () => {
-    if (email==="") {
+    if (email === "") {
       setMessage("Email can't be empty.");
       setVisible(1);
-    }
-    else if (password==="") {
+    } else if (password === "") {
       setMessage("Password can't be empty.");
       setVisible(1);
-    }
-    else {
+    } else {
+      updateLoading(true);
```

Comment:
@ChrisCruze in case login fails the **loading** state needs to be flipped to false. Right now, the loading component stays on forever on login fail.

Reply:
Great catch